### PR TITLE
[iOS] Avoid crash when backdrop filter is null for PlatformViews

### PR DESCRIPTION
**Pull request by cyanglaz**
During certain partial repaint logic, the backdrop filter layer is present but the filter itself is null. We need to account for it when deciding whether to add such filter to the mutator stack.

Fixes https://github.com/flutter/flutter/issues/127095

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: testing/scenario_app/lib/src/platform_view.dart [hunk 1]
```diff
@@ -1443,8 +1443,6 @@ class PlatformViewWithOtherBackDropFilter extends PlatformViewScenario {
 /// A simple platform view for testing backDropFilter with a platform view in the scene.
 ///
 /// The stack would look like: picture 1 -> pv1 -> picture 2 -> filter -> pv2 - > picture 3.
-/// Because backdrop filter on platform views has not been implemented(see: https://github.com/flutter/flutter/issues/43902),
```
Comment (cyanglaz) [inline testing/scenario_app/lib/src/platform_view.dart:1446]: Removing this out dated comment, the backdrop filter on platform views have been implemented.

**Comment by cyanglaz** [comment]
@hellohuanlin Friendly ping :)

File: testing/scenario_app/lib/src/platform_view.dart [hunk 2]
```diff
@@ -1533,6 +1531,68 @@ class TwoPlatformViewsWithOtherBackDropFilter extends Scenario
   }
 }
 
+/// A simple platform view for testing backDropFilter with a platform view in the scene.
+///
+/// The backdrop filter sigma value is negative, which tries to reproduce a crash, see:
```
Comment (hellohuanlin) [inline testing/scenario_app/lib/src/platform_view.dart:1536]: curious what does a negative sigma mean? is it not supposed to happen? maybe make sense to print a warning if so.

**Review by hellohuanlin — APPROVED** [review]

File: testing/scenario_app/lib/src/platform_view.dart (hunk 2 above)
Comment (cyanglaz) [reply testing/scenario_app/lib/src/platform_view.dart:1536]: I'm not entirely sure why the negative sigma is used in one of the example code to reproduce the bug, however, the fix is more about accounting the possibility where the filter object is null. 

As a follow up, I think we also need to disable partial repaint when there is a platform view.

**Comment by sang-makebetter** [comment]
@hellohuanlin Hello, can I ask on which Flutter release that will apply these changes. Thank you!

**Comment by ChurikiTenna** [comment]
Is this fixed yet?
My flutter version is Flutter 3.10.6 • channel stable, and still seeing the crash.

**Pull request merged**